packages/golden-test-generator/src/AddressBalance/getOnChainAddressBalances.ts

```diff
@@ -47,7 +50,8 @@ export const getOnChainAddressBalances = (
           compactGenesis: await StateQuery.genesisConfig(
             await createInteractionContext(reject, logger.info, { connection: options.ogmiosConnectionConfig })
           ),
-          intersection: undefined
+          // Review: this can't be undefined acccording to type
```

Comment:
We could type the constant as a partial, then coerce it after the intersection is set?
``` TypeScript
const response: Partial<AddressBalancesResponse> = {
...
if (atBlocks.includes(currentBlock)) {
  response.balances[currentBlock] = { ...trackedAddressBalances };
    if (atBlocks[atBlocks.length - 1] === currentBlock) {
      draining = true;
      await syncClient.shutdown();
      return resolve(response as ChainSync.Intersection);
    }
  }
```

Reply:
I tried that and it wasn't pretty. I think leaving the double cast is best there.